src/rockstor/fs/btrfs.py

```diff
@@ -1002,8 +1010,39 @@ def balance_status(pool):
     return stats
 
 
-def device_scan():
-    return run_command([BTRFS, 'device', 'scan'])
+def device_scan(dev_byid_list=['all']):
+    """
+    When called with no parameters a 'btrfs device scan' is executed, ie a
+    system wide scan of all /dev block devices to update their btrfs status.
+    Otherwise the list of devices is iterated and a 'btrfs device scan dev'
+    is executed for each item in the passed list. Detached device names and
+    path names that don't exist are ignored.
+    :param dev_byid_list: list of byid device names (without paths) to perform
+    a 'btrfs device scan' on. If not supplied then a single element list
+    ['all'] is substituted and this flags a system wide scan request.
+    :return: (out, err, rc) of the first rc !=0 run or the last rc = 0 run.
+    """
+    out = err = ['']
+    # default to successful return code unless we find otherwise.
+    rc = 0
+    if len(dev_byid_list) > 0:
+        if dev_byid_list[0] == 'all':
+            return run_command([BTRFS, 'device', 'scan'])
+        for dev_byid in dev_byid_list:
+            if re.match('detached-', dev_byid) is not None:
+                # Skip detached devices as we know they don't exist.
+                # Potential log point for early detached device discovery.
+                continue
+            dev_byid_withpath = ('/dev/disk/by-id/%s' % dev_byid)
+            if os.path.exists(dev_byid_withpath):  # only scan existing devices
+                out, err, rc = run_command(
+                    [BTRFS, 'device', 'scan', dev_byid_withpath])
```

Comment:
@schakrava Thanks for the review.

I think I see what you mean however what about the prior and current use of device_scan() in main() of bootstrap.py. if we silence the exception will it not cause us to fail to catch what is described there (in the exception clause) as "... a serious problem." where upon we sys.exit(1) from bootstrap. Like wise when in development I called a partitioned device (by it's root dev name) the error was surfaced to the user, via mount_root(pool). Hence the accompanying target_name property additions in this pr, ie the surfacing of the error led to identifying the need for target_name.

However I'm a bit vague on how the exceptions cascade actually, but it looked like we needed this one to not be silenced in order to identify/surface larger problems so I left it 'as was' with no throw=False, guided in part by the existing device_scan() prior to this pr and it's use in bootstrap.py and the behaviour as described below:

I.e if we revert the disk.target_name to disk.name in mount_root():

    device_scan([dev.target_name for dev in pool.disk_set.all()])

then 'as is' (without throw=False) we get the following when creating a pool with a single partition as the only device member:

![dev-scan-partitioned-by-root](https://user-images.githubusercontent.com/2521585/27488601-813457f0-582f-11e7-8d1f-632e90617a97.png)

because we call btrfs dev scan virtio-serial-1 instead of the intended virtio-serial-1-part2. There by indicating/surfacing the failed call and in turn the incorrect device name use.

However if we do the same but with throw=False added as indicated then we have a silent failure of an incorrect "btrfs dev scan virtio-serial-1" system call.

Apologies if I have missed you point here.